models/POP/shell_scripts/cesm2_1/assimilate.csh

```diff
@@ -0,0 +1,358 @@
+      else
+         echo "ERROR: Requested PRIOR inflation but specified no incoming inflation SD file."
+         echo "ERROR: expected something like ${CASE}.pop.output_priorinf_sd.YYYY-MM-DD-SSSSS.nc"
+         exit 2
+      endif
+
+   endif
+else
+   echo "Prior Inflation           not requested for this assimilation."
+endif
+
+# POSTERIOR: We look for the 'newest' and use it - IFF we need it.
+
+if ( $POSTE_INF > 0 ) then
+
+   if ($POSTE_TF == false) then
+      # we are not using an existing inflation file.
+      echo "inf_flavor(2) = $POSTE_INF, using namelist values."
+
+   else if ( -e pop_inflation_cookie ) then
+      # We want to use an existing inflation file, but this is
+      # the first assimilation so there is no existing inflation
+      # file. This is the signal we need to to coerce the namelist
+      # to have different values for this execution ONLY.
+      # Since the local namelist comes from CASEROOT each time, we're golden.
+
+      set POSTE_TF = FALSE
+
+ex input.nml <<ex_end
+g;inf_initial_from_restart ;s;= .*;= .${PRIOR_TF}., .${POSTE_TF}.,;
+g;inf_sd_initial_from_restart ;s;= .*;= .${PRIOR_TF}., .${POSTE_TF}.,;
+wq
+ex_end
+
+   else
+      # Look for the output from the previous assimilation
+      # Checking for a posterior inflation file to use
+
+      (ls -rt1 ${CASE}.pop.output_postinf_mean.* | tail -n 1 >! latestfile) > & /dev/null
+      set nfiles = `cat latestfile | wc -l`
+
+      if ( $nfiles > 0 ) then
+         set latest = `cat latestfile`
+         ${LINK} $latest input_postinf_mean.nc
+      else
+         echo "ERROR: Requested POSTERIOR inflation but specified no incoming inflation file."
+         echo "ERROR: expected something like ${CASE}.pop.output_postinf_mean.YYYY-MM-DD-SSSSS.nc"
+         exit -5
+      endif
+
+      # Checking for a posterior inflation sd file to use
+
+      (ls -rt1 ${CASE}.pop.output_postinf_sd.* | tail -n 1 >! latestfile) > & /dev/null
+      set nfiles = `cat latestfile | wc -l`
+
+      if ( $nfiles > 0 ) then
+         set latest = `cat latestfile`
+         ${LINK} $latest input_postinf_sd.nc
+      else
+         echo "ERROR: Requested POSTERIOR inflation but specified no incoming inflation SD file."
+         echo "ERROR: expected something like ${CASE}.pop.output_postinf_sd.YYYY-MM-DD-SSSSS.nc"
+         exit 2
+      endif
+
+   endif
+
+else
+   echo "Posterior Inflation       not requested for this assimilation."
+endif
+
+#=========================================================================
+# Block 5: REQUIRED DART namelist settings
+#=========================================================================
+${REMOVE} restarts_in.txt restarts_out.txt
+
+foreach FILE ( rpointer.ocn_????.restart )
+   head -n 1 ${FILE} >> restarts_in.txt
+end
+
+# WARNING: this is the part where the files just get overwritten
+${COPY} restarts_in.txt restarts_out.txt
+
+# POP always needs a pop_in and a pop.r.nc to start.
```

Comment:
It's actually filter than needs a pop_in and a pop.r.nc to start.

Reply:
Fixed the comment.